packages/SwingSet/src/controller.js

```diff
@@ -125,15 +125,47 @@ export function loadBasedir(basedir) {
   return config;
 }
 
+/**
+ * Resolve a pathname found in a config descriptor.  First try to resolve it as
+ * a module path, and then if that doesn't work try to resolve it as an
+ * ordinary path relative to the directory in which the config file was found.
+ *
+ * @param dirname  Path to directory containing the config file
+ * @param specPath  Path found in a `sourceSpec` or `bundleSpec` property
+ *
+ * @return the absolute path corresponding to `specPath` if it can be
+ *    determined.
+ */
+function resolveSpecFromConfig(dirname, specPath) {
+  try {
+    return require.resolve(specPath, { path: [dirname] });
+  } catch (e) {
+    if (e.code !== 'MODULE_NOT_FOUND') {
+      throw e;
+    }
+  }
+  return path.resolve(dirname, specPath);
```

Comment:
Okay, now I see what you're going for.  I, personally, would prefer that a specifier is the exact same resolution semantics as for modules (i.e. use `./foo.js` if you mean `foo.js` in the current directory).  If you agree with that sentiment, you could print a deprecation warning if the `path.resolve` is hit.

Up to you, though.  I can see the convenience of being able to resolve bare specifiers relative to dirname, even though it gives me the willies. 😄

Reply:
Would it maybe be better to use `path.resolve()` first on bare specifiers, then punt to `require.resolve` for everything else or if the resolve doesn't result in a file?  That would eliminate the surprise when you do `"lodash"` but you meant `"./lodash/index.js"`.

Reply:
Keep in mind that these paths appear inside config files and have a different use pattern from imports.  With imports, it's common that a large fraction of the stuff you are referring to comes from someplace else.  In config files, non-local references are the exception; you would never reference `"./lodash/index.js"`.  Not having to stick a redundant `./` in front of almost literally everything improves readability quite a bit.